src/Compilers/CSharp/Test/Emit3/Semantics/ExtensionTests2.cs

```diff
@@ -2285,5 +2285,47 @@ public static class E
             //         [System.Diagnostics.CodeAnalysis.UnscopedRef]
             Diagnostic(ErrorCode.ERR_UnscopedRefAttributeUnsupportedMemberTarget, "System.Diagnostics.CodeAnalysis.UnscopedRef").WithLocation(8, 10));
     }
+
+    [Fact]
+    public void Nameof_AllowStaticInstanceMismatch()
+    {
+        var src = """
+public static class E
+{
+    extension(object o)
+    {
+        [My(nameof(string.P)), My(nameof(E.get_P)), My(nameof(E.M))]
+        [My(nameof(string.M))]
+        public void M() { }
+
+        public int P => 42;
+    }
+
+    extension(string)
+    {
+        [My(nameof(string.P)), My(nameof(E.get_P)), My(nameof(E.M))]
+        [My(nameof(string.M))]
+        public static void M() { }
+
+        public static int P => 42;
+    }
+}
+
+[System.AttributeUsage(System.AttributeTargets.All, AllowMultiple = true)]
+public class MyAttribute : System.Attribute
+{
+    public MyAttribute(string s) { }
+}
+""";
+
+        var comp = CreateCompilation(src);
+        comp.VerifyEmitDiagnostics(
+            // (6,20): error CS8093: Extension method groups are not allowed as an argument to 'nameof'.
```

Comment:
>// (6,20): error CS8093: Extension method groups are not allowed as an argument to 'nameof'. [](http://example.com/codeflow?start=12&length=92)

Given that ```nameof``` cannot refer to an extension method. It is not clear to me why it should work with extension properties.

Reply:
Because that is the only way you can do a `nameof` to get the name of an extension property.
For extension methods, you can just use the static form: `nameof(E.M)`.

I didn't end up bringing this [question](https://github.com/dotnet/csharplang/blob/main/proposals/extensions.md#nameof) to LDM, but I can confirm next week if you feel that's useful.

Reply:
I think this is worth a discussion at LDM. For example, if we allow this for properties, I think we should drop the restriction for methods as well. At the same time, if we felt that it is necessary to protect users from something with methods, the same reasoning should be applied to properties as well.